src/app/DIAmeterApplication.cpp

```diff
@@ -0,0 +1,558 @@
+#include <cstdio>
+#include "app/tide/abspath.h"
+#include "app/tide/records_to_vector-inl.h"
+
+#include "io/carp.h"
+#include "parameter.h"
+#include "io/SpectrumRecordWriter.h"
+#include "TideIndexApplication.h"
+#include "TideSearchApplication.h"
+#include "DIAmeterApplication.h"
+#include "ParamMedicApplication.h"
+#include "PSMConvertApplication.h"
+#include "tide/mass_constants.h"
+#include "TideMatchSet.h"
+#include "util/Params.h"
+#include "util/FileUtils.h"
+#include "util/StringUtils.h"
+
+
+const double DIAmeterApplication::XCORR_SCALING = 100000000.0;
+const double DIAmeterApplication::RESCALE_FACTOR = 20.0;
+
+DIAmeterApplication::DIAmeterApplication() { /* do nothing */ }
+DIAmeterApplication::~DIAmeterApplication() { /* do nothing */ }
+
+
+int DIAmeterApplication::main(int argc, char** argv) {
+  return main(Params::GetStrings("tide spectra file"));
+}
+int DIAmeterApplication::main(const vector<string>& input_files) {
+  return main(input_files, Params::GetString("tide database"));
+}
+
+int DIAmeterApplication::main(const vector<string>& input_files, const string input_index) {
+  carp(CARP_INFO, "Running tide-search...");
+
+  const string index = input_index;
+  string peptides_file = FileUtils::Join(index, "pepix");
+  string proteins_file = FileUtils::Join(index, "protix");
+  string auxlocs_file = FileUtils::Join(index, "auxlocs");
+
+  // params
+  double bin_width_  = Params::GetDouble("mz-bin-width");
+  double bin_offset_ = Params::GetDouble("mz-bin-offset");
+  vector<int> negative_isotope_errors = TideSearchApplication::getNegativeIsotopeErrors();
+  // bool use_neutral_loss_peaks_ = Params::GetBool("use-neutral-loss-peaks");
+  // bool use_flanking_peaks_ = Params::GetBool("use-flanking-peaks");
+
```

Comment:
throw an error if tailor calibration is False

Reply:
No need. I will use tailed-calibrated-xcorr regardless. It's not an option to DIAmeter

Reply:
No need. I will use tailed-calibrated-xcorr regardless. It's not an option to DIAmeter